src/git/github.js

```diff
@@ -19,17 +19,26 @@ const parsePath = require('parse-filepath');
 
 module.exports = class Github extends Git {
   constructor(reqOpt) {
-    super(reqOpt);  
+    super(reqOpt);
+    if (this.host == 'github.com') {
+      this._enterprise = 'https://api.github.com';
+    } else {
+      this._enterprise = `http://${this.host}/api/v3`;
+    }
+  }
+
+  get enterprise() {
+    return this._enterprise;
   }
 
   getReqUrl() {
-    let enterprise = '';
-    if (this.host == 'github.com') {
-      enterprise = 'https://api.github.com';
+    if (this.release) {
+      return `${this.enterprise}/repos/${this.repo}/releases/tags/${this.release}`;
+    } else if (this.ref) {
+      return `${this.enterprise}/repos/${this.repo}/contents/${this.path}?ref=${this.ref}`;
     } else {
-      enterprise = `http://${this.host}/api/v3`;
+      return `${this.enterprise}/repos/${this.repo}/contents/${this.path}`;
```

Comment:
Maybe add a comment to this else  
// Use the default branch if no release or branch specified